ksqldb-engine/src/test/java/io/confluent/ksql/materialization/ks/KsMaterializationFunctionalTest.java

```diff
@@ -745,5 +756,21 @@ private static void initializeKsql(final TestKsqlContext ksqlContext) {
         + ");"
     );
   }
+
+  /*
+   * table.get() calls can sometimes fail when partitions are being rebalanced.
+   * When that happens: retry!
+   */
+  private static <T> T withRetry(final Supplier<T> supplier) {
```

Comment:
It might be nice to pull this retry logic out to a common utility.  This is a common pattern, though not sure if I've seen it elsewhere in KSQL.

Reply:
done. I've extended the existing `assertThatEventually`.